experimental/packages/opentelemetry-instrumentation-fetch/test/fetch.test.ts

```diff
@@ -235,28 +244,22 @@ describe('fetch', () => {
         } else {
           input.text().then(r => (requestBody = r));
         }
-
-        if (init instanceof Request) {
-          // Passing request as 2nd argument causes missing body bug (#2411)
-          response.status = 400;
-          response.statusText = 'Bad Request (Request object as 2nd argument)';
-          reject(new window.Response(JSON.stringify(response), response));
-        } else if (init.method === 'DELETE') {
-          response.status = 405;
-          response.statusText = 'OK';
-          resolve(new window.Response('foo', response));
+        if (init.method === 'DELETE') {
+          responseInit.status = 405;
+          responseInit.statusText = 'OK';
+          responseBody = 'foo';
         } else if (
           (input instanceof Request && input.url === url) ||
           input === url
         ) {
-          response.status = 200;
-          response.statusText = 'OK';
-          resolve(new window.Response(JSON.stringify(response), response));
+          responseInit.status = 200;
+          responseInit.statusText = 'OK';
         } else {
-          response.status = 404;
-          response.statusText = 'Bad request';
-          reject(new window.Response(JSON.stringify(response), response));
+          responseInit.status = 404;
+          responseInit.statusText = 'Not found';
         }
+
+        resolve(new window.Response(responseBody, responseInit));
```

Comment:
We now always resolve, as the real `fetch()` would. The 404 branch is also never exercised in the tests (as evident by the fact that I changed it from `resolve()` to `reject()` and nothing broke). As mentioned above, rejecting with a response is never correct.

If we are rejecting here because it's an internal assertion that indicating some test is set up wrong, I'd rather see this reject with an regular Error, or just throw synchronously at the top.